src/modules/mc_pos_control/mc_pos_control_main.cpp

```diff
@@ -1694,14 +1709,14 @@ MulticopterPositionControl::calculate_velocity_setpoint(float dt)
 	limit_altitude();
 
 	if (_run_alt_control) {
-		_vel_sp(2) = (_pos_sp(2) - _pos(2)) * _params.pos_p(2);
-	}
-
-	/* make sure velocity setpoint is saturated in xy*/
-	float vel_norm_xy = sqrtf(_vel_sp(0) * _vel_sp(0) +
-				  _vel_sp(1) * _vel_sp(1));
+		if (PX4_ISFINITE(_pos_sp(2))) {
+			_vel_sp(2) = (_pos_sp(2) - _pos(2)) * _params.pos_p(2);
 
-	slow_land_gradual_velocity_limit();
```

Comment:
You just kicked this out without replacement. I created a nicer replacement of this but it's not integrated yet.